packages/core/src/core/client.ts

```diff
@@ -271,6 +275,10 @@ export class GeminiClient {
     turns: number = this.MAX_TURNS,
     originalModel?: string,
   ): AsyncGenerator<ServerGeminiStreamEvent, Turn> {
+    if (this.lastPromptId !== prompt_id) {
```

Comment:
We'll also want to reset when you don't get aggregate tool calls. For instance, it's very common to see ReadFile(A.txt) over and over again (very validly) inbetween edits to accomplish an end-task.